docs/pm-doc.md

```diff
@@ -0,0 +1,682 @@
+echo 'export PATH=$PATH:$HOME/bin' >> ~/.bashrc
+```
+
+Installing `odo` on `s390x` architecture:
+
+1. Download the latest nightly build:
+```shell
+curl -L https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-nightly-builds/odo-linux-s390x -o odo
+```
+
+2. Install odo:
+```shell
+sudo install -o root -g root -m 0755 odo /usr/local/bin/odo
+```
+
+3. (Optional) If you do not have root access, you can install `odo` to the local directory and add it to your `$PATH`:
+
+```shell
+mkdir -p $HOME/bin 
+cp ./odo $HOME/bin/odo
+export PATH=$PATH:$HOME/bin
+# (Optional) Add the $HOME/bin to your shell initialization file
+echo 'export PATH=$PATH:$HOME/bin' >> ~/.bashrc
+```
+
+---
+
+#### MacOS
+
+Installing `odo` on `amd64` architecture:
+
+1. Download the latest nightly build:
+```shell
+curl -L https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-nightly-builds/odo-darwin-amd64 -o odo
+```
+
+2. Install odo:
+```shell
+chmod +x ./odo
+sudo mv ./odo /usr/local/bin/odo
+```
+
+3(Optional) If you do not have root access, you can install `odo` to the local directory and add it to your `$PATH`:
+
+```shell
+mkdir -p $HOME/bin 
+cp ./odo $HOME/bin/odo
+export PATH=$PATH:$HOME/bin
+# (Optional) Add the $HOME/bin to your shell initialization file
+echo 'export PATH=$PATH:$HOME/bin' >> ~/.bashrc
+```
+
+Installing `odo` on `arm64` architecture:
+
+1. Download the latest nightly build:
+```shell
+curl -L https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-nightly-builds/odo-darwin-arm64 -o odo
+```
+
+2. Install odo:
+```shell
+chmod +x ./odo
+sudo mv ./odo /usr/local/bin/odo
+```
+
+3. (Optional) If you do not have root access, you can install `odo` to the local directory and add it to your `$PATH`:
+
+```shell
+mkdir -p $HOME/bin 
+cp ./odo $HOME/bin/odo
+export PATH=$PATH:$HOME/bin
+# (Optional) Add the $HOME/bin to your shell initialization file
+echo 'export PATH=$PATH:$HOME/bin' >> ~/.bashrc
+```
+
+---
+
+#### Windows
+
+1. Open a PowerShell terminal
+
+2. Download the latest nightly build:
+```shell
+curl -L https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-nightly-builds/odo-windows-amd64.exe -o odo.exe
+```
+
+3. Add the binary to your `PATH`
+
+## Developing with Java (Spring Boot)
+
+### Step 0. Creating the initial source code (optional)
+
+We will create the example source code by using some popular frameworks.
+
+Before we begin, we will create a new directory and cd into it.
+```shell
+mkdir quickstart-demo && cd quickstart-demo
+```
+
+This is *optional* and you may use an existing project instead (make sure you cd into the project directory before running any odo commands) or a starter project from `odo init`.
+
+For Java, we will use the [Spring Initializr](https://start.spring.io/) to generate the example source code:
+
+1. Navigate to [start.spring.io](https://start.spring.io/).
+2. Select **Maven** under **Project**.
+3. Under **Spring Boot** select **2.7.\*** version.
+:::caution
```

Comment:
I guess we'll also need to replace these admonitions (syntax specific to Docusaurus, I think). Maybe we can also replace those with something else like simple blockquotes?